hudi-common/src/main/java/org/apache/hudi/internal/schema/visitor/NameToPositionVisitor.java

```diff
@@ -34,65 +30,36 @@
 /**
  * Schema visitor to produce name -> position map for internalSchema. Positions are assigned in a depth-first manner.
  */
-public class NameToPositionVisitor extends InternalSchemaVisitor<Map<String, Integer>> {
-  private final Deque<String> fieldNames = new LinkedList<>();
-  private final Map<String, Integer> nameToId = new HashMap<>();
+public class NameToPositionVisitor extends AbstractNameVisitor<Map<String, Integer>> {
   private final AtomicInteger position = new AtomicInteger(0);
+  public NameToPositionVisitor() {
+    super(new HashMap<>());
+  }
 
   @Override
   public void beforeField(Types.Field field) {
     nameToId.put(createFullName(field.name(), fieldNames), position.getAndIncrement());
     fieldNames.push(field.name());
   }
 
-  @Override
-  public void afterField(Types.Field field) {
```

Comment:
This will cause code to read a little difficulty.
In the original implementation, can read the `afterField` after `beforeField`. now, if I am author for this code, when I update the code, I know there has one method is `afterField` also need to change, but for other developer, may lose this method.
And has this problem when reading the code too, hese methods are not suitable for independent abstraction in a sense

Reply:
I think we can keep the current code for now.  @Davis-Zhang-Onehouse lmk if we can close this PR.